src/main/java/seedu/address/logic/commands/SortCommand.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Sorts and lists all persons in the address book by the specified criteria.
+ * The available sorting criteria are by name or by appointment.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts all persons by the specified criteria.\n"
+            + "Parameters: name | appointment\n"
+            + "Example: " + COMMAND_WORD + " name\n"
+            + "         " + COMMAND_WORD + " appointment\n";
+
+    public static final String MESSAGE_INVALID_SORTING_CRITERIA = "Invalid sorting criteria.\n";
```

Comment:
maybe this is more suited to be in the Messages class instead to reduce clutter and coupling?